graphstorm-processing/tests/test_dist_heterogenous_loader.py

```diff
@@ -362,6 +406,95 @@ def test_load_dist_hgl_without_labels(
     assert metadata["edge_data"] == expected_edge_data
 
 
+def test_write_edge_structure_homogeneous_reverse_edges(
+    spark: SparkSession,
+    data_configs_homogeneous,
+    dghl_loader_homogeneous: DistHeterogeneousGraphLoader,
+    user_rated_movie_df: DataFrame,
+    tempdir,
+):
+    edge_configs = data_configs_homogeneous["edges"]
+
+    # We need these two for the side-effects of creating the mappings
+    missing_node_types = dghl_loader_homogeneous._get_missing_node_types(edge_configs, [])
+    dghl_loader_homogeneous.create_node_id_maps_from_edges(edge_configs, missing_node_types)
+
+    edge_dict: Dict[str, Dict] = {
+        "data": {
+            "format": "csv",
+            "files": ["edges/user-rated-movie.csv"],
+            "separator": ",",
+        },
+        "source": {"column": "~from", "type": "_N"},
+        "relation": {"type": "_E"},
+        "dest": {"column": "~to", "type": "_N"},
+    }
+
+    edge_config = EdgeConfig(edge_dict, edge_dict["data"])
+    _, path_list, rev_edge_path_list = dghl_loader_homogeneous.write_edge_structure(
+        user_rated_movie_df, edge_config
+    )
+
+    assert len(rev_edge_path_list) == 0
+
+    edge_file_path = [
+        os.path.join(dghl_loader_homogeneous.local_meta_output_path, x) for x in path_list
+    ]
+    edge_df = spark.read.parquet(*edge_file_path)
+
+    # Only one node type in the homogeneous graph
+    assert len(dghl_loader_homogeneous.node_mapping_paths) == 1
+    for node_type, mapping_files in dghl_loader_homogeneous.node_mapping_paths.items():
+        files_with_prefix = [
+            os.path.join(dghl_loader_homogeneous.local_meta_output_path, x) for x in mapping_files
+        ]
+        mapping_df = spark.read.parquet(*files_with_prefix)
+
+    # Mapping source id to int
+    user_rated_movie_df_int_id = user_rated_movie_df.join(
+        mapping_df, F.col("`~from`") == F.col("orig"), "inner"
+    ).select(F.col("new").alias("src_int_id"), F.col("~to"))
+    # Mapping dest id to int
+    user_rated_movie_df_int_id = user_rated_movie_df_int_id.join(
+        mapping_df, F.col("`~to`") == F.col("orig"), "inner"
+    ).select(F.col("src_int_id"), F.col("new").alias("dst_int_id"))
+    user_rated_movie_df_int_id_reverse = user_rated_movie_df_int_id.select(
+        F.col("dst_int_id").alias("src_int_id"), F.col("src_int_id").alias("dst_int_id")
+    )
+    user_rated_movie_df_int_id = user_rated_movie_df_int_id.union(
+        user_rated_movie_df_int_id_reverse
+    )
+    processing_result_row = set(edge_df.collect())
+    expected_edge_row = set(user_rated_movie_df_int_id.collect())
```

Comment:
Is this duplicating the implementation? What are we trying to test?

Reply:
This is about we are trying to ensure the reversing edges df will be the same as we expected. 

* There is no edge_type-rev generated
* The reverse edges are added to the same edge type.